### ISSUE TITLE
Logging connection as Disposed before actually being disposed

### ISSUE THREAD

**Issue by OskarKlintrot** [aspnet/EntityFramework6#462]
To me it looks as if the `DatabaseLogFormatter` is calling `Write` with the string `Disposed connection at {0}{1}` when the connection will be disposed and not when it actually has been disposed: https://github.com/aspnet/EntityFramework6/blob/master/src/EntityFramework/Infrastructure/Interception/DatabaseLogFormatter.cs#L676-L694

**Comment by ajcvickers** [comment]
@OskarKlintrot Can you be more specific about what issues is this causing for you?

**Comment by OskarKlintrot** [comment]
I was just looking at the `DatabaseLogFormatter` to implement my own and noticed that in the method that gets called before disposing the connection the message says that it has already been disposed. The method that gets called after the connection actually has been disposed is empty: https://github.com/aspnet/EntityFramework6/blob/master/src/EntityFramework/Infrastructure/Interception/DatabaseLogFormatter.cs#L696-L703

So in short, nothing worse than a bit of confusion.
